### PR TITLE
Adds a mailbox/post box

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Adds a mailbox. which spawns outside cargo in maps. Anyone can put mail into it, and it gets "sorted" into the persons box. If you have cargo access, you can open anyones box (if you want to deliver by hand), but otherwise you can only access your own box. 

## Why It's Good For The Game
Helps streamline cargo/mail greatly, makes it so much easier to walk up, check whether you have mail, then be on your way.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
image
image
image

</details>

## Changelog
:cl:
add: mailbox
/:cl:

### CODE REVIEW

File: code/game/objects/items/mail_box.dm [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	Mailboxes/P.O. Boxes! (idk which one fits more)
+	Contains mail, using it with an empty hand opens a menu for you to pull out your own mail.
+	Having cargo bay access lets you pull out anyone's mail, so it doesn't get trapped in limbo
+
+	made by candycane/etherware
+
+*/
+
+
+/obj/machinery/mailbox
```
Comment (Sarchutar) [inline code/game/objects/items/mail_box.dm]: why is it a machine? couldn't it just be a structure?

File: code/game/objects/items/mail_box.dm [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	name = "official postbox"
+	desc = "A small postbox stationed for easy access, containing station letters."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "mailbox"
+	density = TRUE
+	use_power = 0
```
Comment (Sarchutar) [inline code/game/objects/items/mail_box.dm]:
```suggestion
	use_power = NO_POWER_USE
```
consistency

File: code/game/objects/items/mail_box.dm [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	idle_power_usage = 0
+	active_power_usage = 0
+	power_channel = AREA_USAGE_EQUIP
```
Comment (Sarchutar) [inline code/game/objects/items/mail_box.dm]: those three are same as the parent and can be removed

File: code/game/objects/items/mail_box.dm [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	if(active_slots[user.real_name])
+		. += "<span class='notice'>You spot a green light above your tray!</span>"
+
+/obj/machinery/mailbox/attack_hand(mob/living/carbon/user)
+	if(!length(active_slots))
+		to_chat(user, "<span class='notice'>All of the trays are empty!</span>")
+		return
+	var/datum/mail_slot/personal_slot
+	if(allowed(user) && active_slots)
+		var/manual_name = input("Who's mail would you like to access?", "Mail", null) as null|anything in active_slots
```
Comment (Sarchutar) [inline code/game/objects/items/mail_box.dm:36]: return if nothing was selected

File: code/game/objects/items/mail_box.dm [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+			inserting_mail.forceMove(src)
+			to_chat(user, "<span class='notice'>You slip the mail into the spam box.</span>")
+			return
+
+		var/datum/mind/recipient = inserting_mail.recipient_ref.resolve()
+		if(!recipient)
+			return
+
+		var/mail_to = recipient.name
+		to_chat(user, personal_slot)
```
Comment (Sarchutar) [inline code/game/objects/items/mail_box.dm]: leftover debug message?

File: code/game/objects/items/mail_box.dm [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+			to_chat(user, "<span class='warning'>You can't find a box!</span>")
+			return
+
+		to_chat(user, "<span class='notice'>You slip the mail into the proper box.</span>")
+		personal_slot.insert_mail(I)
+
+/datum/mail_slot
+	/// the name of the person able to open this
+	var/list/mail_stack = list()
+	var/locked = FALSE
```
Comment (Sarchutar) [inline code/game/objects/items/mail_box.dm:101]: `var/locked` is unused

File: code/game/objects/items/mail_box.dm (hunk 1 above)
Comment (etherware-novice) [reply code/game/objects/items/mail_box.dm]: oh

File: code/game/objects/items/mail_box.dm (hunk 5 above)
Comment (etherware-novice) [reply code/game/objects/items/mail_box.dm]: yes

**Comment by FriendlyContractor** [comment]
> Helps streamline cargo/mail greatly, makes it so much easier to walk up, check whether you have mail, then be on your way

We want exactly the opposite of what you are suggesting. Players need to be forced to interact with each other, **NOT** given more conveniences so they can avoid others. You are trying to ruin everything good about mail system.

File: code/game/objects/items/mail_box.dm [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	desc = "A small postbox stationed for easy access, containing station letters."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "mailbox"
+	density = TRUE
+	max_integrity = 1000
+	req_access = list(ACCESS_CARGO)  // can look thru anyones mail
+	var/list/active_slots = list()
+	var/list/spam_mail = list()
+
+/obj/machinery/mailbox/examine(mob/user)
```
Comment (Sarchutar) [inline code/game/objects/items/mail_box.dm:23]: you do actually have to change every `/obj/machinery/mailbox` to `/obj/structure/mailbox`. Also don't forget the maps

Comment (etherware-novice) [reply code/game/objects/items/mail_box.dm:23]: oops

**Comment by itsmeow** [comment]
:key:
The entire fun of the mail system is the delivery part. This effectively bypasses that and makes the job of the mailman useless, now it's just a "put mail crate into mailbox, tell everyone mail is here on radio", instead of a "be a mailman, track people down and deliver mail ruthlessly".

**Comment by PestoVerde322** [comment]
As it is right now, i quote Itsmeow, where basically you're removing the only fun in mail delivering. However, in the future, i have a use case use for this, which is mail deposit for station to outside mail, but as it is right now, it doesn't really have much need.

**Comment by etherware-novice** [comment]
alright ill move it to outside departments

**Comment by itsmeow** [comment]
I still don't think that is in the spirit of the mailman. The fun is hunting down the individual.

**Comment by EvilDragonfiend** [comment]
> alright ill move it to outside departments

Regardless it's inside or outside, you will dump mails into it.

**Comment by Penwin0** [comment]
I agree with the reasoning posted by the other maintainers. This trivialized the interaction to the maximum extent

**Pull request closed without merging**